src/narjure/bag.clj

```diff
@@ -2,18 +2,33 @@
   (:require [avl.clj :as avl]))
 
 (defn compare-elements
+  "Compares elements. If priority of elements is equal, compares the hashes of
+  the ids. It is done to allow bag to contain elements with equal priority."
   [{p1 :priority id1 :id}
    {p2 :priority id2 :id}]
   (if (= p1 p2)
     (> (hash id1) (hash id2))
     (and (> p1 p2) (not= id1 id2))))
 
 (defprotocol Bag
-  (add-element [_ element])
-  (get-by-index [_ index])
-  (get-by-id [_ id])
-  (pop-element [_])
-  (update-element [_ element])
+  (add-element
+    ;Adds element to bag, removes element with lowest priority if bag is full.
+    [_ element])
+  (get-by-index
```

Comment:
get-by-index - it is unexpected that an index could return nil. This doesn't happen in curve bag. What's the frequency of nil likely to be?